src/main/java/bio/terra/service/snapshot/flight/delete/DeleteOutstandingSnapshotAccessRequestsStep.java

```diff
@@ -0,0 +1,56 @@
+package bio.terra.service.snapshot.flight.delete;
+
+import bio.terra.common.exception.NotFoundException;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.model.EnumerateSnapshotAccessRequest;
+import bio.terra.service.snapshotbuilder.SnapshotBuilderService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DeleteOutstandingSnapshotAccessRequestsStep implements Step {
+  private static final Logger logger =
+      LoggerFactory.getLogger(DeleteOutstandingSnapshotAccessRequestsStep.class);
+  private final SnapshotBuilderService snapshotBuilderService;
+  private final UUID snapshotId;
+  private final AuthenticatedUserRequest userReq;
+
+  public DeleteOutstandingSnapshotAccessRequestsStep(
+      AuthenticatedUserRequest userReq,
+      UUID snapshotId,
+      SnapshotBuilderService snapshotBuilderService) {
+    this.snapshotId = snapshotId;
+    this.userReq = userReq;
+    this.snapshotBuilderService = snapshotBuilderService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    try {
+      EnumerateSnapshotAccessRequest requestResponseList =
+          snapshotBuilderService.enumerateRequestsBySnapshot(snapshotId);
+      requestResponseList
+          .getItems()
+          .forEach(
```

Comment:
This makes me uneasy that we could run into an NPE here, but I can't quite put my finger on the issue because in theory we should always return at least an empty list from the enumerate result.

Reply:
I could wrap it in an `Objects.requireNotNull()`, but yeah, I think we should always get an empty list.